services/audit/appender.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package audit
+
+import (
+	"context"
+	"io"
+
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/util/rotatingfilewriter"
+)
+
+type Appender interface {
+	Record(context.Context, *Event)
+	Close() error
+	ReleaseReopen() error
+}
+
+// LogAppender writes an info log entry for every audit event
+type LogAppender struct{}
+
+func (a *LogAppender) Record(ctx context.Context, e *Event) {
+	log.Info(e.Message)
```

Comment:
`"%s", e.Message`

Reply:
No need to format a fixed string.

Reply:
It needs. Otherwise what if the "Message" contains "%s" ?

And I think it needs a proper prefix to make it easily identifiable in plenty of log messages.

Reply:
> And I think it needs a proper prefix to make it easily identifiable in plenty of log messages.

We don't prefix the other logs too.

Reply:
The only log without proper "prefix" or "trait" is `log.Info(stopMark)` in `gitea_uploader_test.go`, for a clear testing purpose.

Searched by `log.Info\([^"]`.

Reply:
`log.Info("Failed authentication attempt for %s from %s", ctx.Doer.Name, ctx.RemoteAddr())`

What's the prefix in this case? Maybe we are talking about different things.
An example audit log would be `"Team %s was added to organization %s."`.

Reply:
Prefix or trait: make it easily identifiable in plenty of log messages.

* `Failed authentication attempt`
* `was added to organization`